src/triage/triage.service.ts

```diff
@@ -1,63 +1,643 @@
+      await this.audit.logAction({
+        userId,
+        operation: 'ALERT_FETCH_FAILED',
+        entityName: 'Alert',
+        actionPerformed: `Failed to fetch alert ${alertId}: ${error.message}`,
+        outcome: 'FAILURE',
+      });
+      throw new InternalServerErrorException('Unable to retrieve alert details');
+    }
+  }
+
+  async convertToCase(alertId: string, convertAlertToCase: ConvertAlertToCase, userId: string, tenantId: string) {
+    const alert = await this.prisma.alert.findUnique({
+      where: { alert_id: alertId },
+    });
+
+    if (!alert) {
+      throw new NotFoundException(`Alert ${alertId} not found`);
+    }
+
+    if (alert.tenant_id !== tenantId) {
+      throw new NotFoundException(`Alert ${alertId} not accessible for this tenant`);
+    }
+
+    if (alert.alert_status === AlertStatus.CLOSED) {
+      throw new BadRequestException(`Alert ${alertId} is already closed`);
+    }
+
+    if (alert.alert_status === AlertStatus.CONVERTED) {
+      throw new BadRequestException(`Alert ${alertId} is already converted to a case`);
+    }
+
+    const casePriority = convertAlertToCase.priority ?? alert.priority;
+    try {
+      const newCase = await this.prisma.case.create({
+        data: {
+          case_creator_user_id: userId,
+          case_owner_user_id: userId,
+          tenant_id: alert.tenant_id,
+          priority: casePriority,
+          status: CaseStatus.DRAFT,
+          parent_id: null,
+          case_type: convertAlertToCase.caseType,
+          case_creation_type: CaseCreationType.MANUAL,
+        },
+      });
+
+      await this.prisma.alert.update({
+        where: { alert_id: alertId },
+        data: {
+          alert_status: AlertStatus.CONVERTED,
+          case_id: newCase.case_id,
+        },
+      });
+
+      await this.audit.logAction({
+        userId,
+        operation: 'ALERT_CONVERTED_TO_CASE',
+        entityName: 'Alert',
+        actionPerformed: `Converted alert ${alertId} to case ${newCase.case_id}`,
+        outcome: 'SUCCESS',
+      });
+
+      return newCase;
+    } catch (error) {
+      this.logger.error(`Failed to create convert alert ${alertId} to case`, error);
+      await this.audit.logAction({
+        userId,
+        operation: 'ALERT_CONVERT_TO_CASE_FAILED',
+        entityName: 'Alert',
+        actionPerformed: `Failed to convert alert ${alertId} to case: ${error.message}`,
+        outcome: 'FAILURE',
+      });
+      throw new InternalServerErrorException('Failed to convert alert to case');
+    }
+  }
+
+  async handleAITriage(alertId: string, dto: SubmitAlertDto, userId: string, tenantId: string) {
+    try {
+      // Story 1G
+      // If confidenceThreshold environment variable is not set, default to 100% → ensures low-confidence predictions always go to investigation.
+      const confidenceThreshold = process.env.CONFIDENCE_THRESHOLD ? Number(process.env.CONFIDENCE_THRESHOLD) : 100;
```

Comment:
Better to use configService when dealing with environment variables. Maintains NestJS best practices.
```suggestion
      const confidenceThreshold = this.configService.get<string>('CONFIDENCE_THRESHOLD') ? Number(this.configService.get<string>('CONFIDENCE_THRESHOLD') : 100;
```